### PR TITLE
core(config): add new PWA category groups

### PR DESCRIPTION
**Pull request by brendankenny**
Starting on #6395

This PR just adds groups to the PWA category and deletes the manifest-short-name-length audit. Feel free to come up with better group descriptions or we can defer, I spent no time on them :)

### CODE REVIEW

**Comment by brendankenny** [comment]
> Feel free to come up with better group descriptions or we can defer

removed descriptions (since not actually in the mocks), which makes this even easier :)

File: lighthouse-core/config/default-config.js [hunk 1]
```diff
@@ -317,22 +325,23 @@ const defaultConfig = {
...
-        {id: 'redirects-http', weight: 2},
-        {id: 'viewport', weight: 2},
-        // Relatively easy checkboxes to tick with minimal value on their own
-        {id: 'service-worker', weight: 1},
-        {id: 'without-javascript', weight: 1},
-        {id: 'splash-screen', weight: 1},
-        {id: 'themed-omnibox', weight: 1},
-        {id: 'content-width', weight: 1},
-        {id: 'manifest-short-name-length', weight: 0},
+        // Fast and Reliable
```
Comment (patrickhulce) [inline lighthouse-core/config/default-config.js]: hey this lines up exactly with "most difficult and critical for good UX" :D

do we need to re-order the lines? it lined up with "fast" and then "reliable" already ;)

File: lighthouse-core/config/default-config.js [hunk 2]
```diff
@@ -225,6 +224,15 @@ const defaultConfig = {
       title: str_(UIStrings.diagnosticsGroupTitle),
       description: str_(UIStrings.diagnosticsGroupDescription),
     },
+    'pwa-fast-reliable': {
+      title: 'Fast and reliable',
```
Comment (patrickhulce) [inline lighthouse-core/config/default-config.js]: could we start with these being `str_` ? :)

File: lighthouse-core/config/default-config.js [hunk 3]
```diff
@@ -317,22 +325,23 @@ const defaultConfig = {
...
+        {id: 'works-offline', weight: 5, group: 'pwa-fast-reliable'},
+        {id: 'load-fast-enough-for-pwa', weight: 7, group: 'pwa-fast-reliable'},
+
+        // Installable
+        {id: 'is-on-https', weight: 2, group: 'pwa-installable'},
+        {id: 'service-worker', weight: 1, group: 'pwa-installable'},
+        {id: 'webapp-install-banner', weight: 3, group: 'pwa-installable'},
+
+        // Engaging
+        {id: 'redirects-http', weight: 2, group: 'pwa-engaging'}, // Arguably not engaging
```
Comment (patrickhulce) [inline lighthouse-core/config/default-config.js]: with the new in-yo-face red X insecure I think it works, I'm always a bit taken aback on surge.sh URLs these days on mobile canary!

File: lighthouse-core/config/default-config.js (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/config/default-config.js]: oh, ha, nm, I see what you mean. SGTM :)

**Pull request merged**